docs/builders/webpack.md

```diff
@@ -0,0 +1,201 @@
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/my-component-story-import-static-asset.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+#### Import JSON as JavaScript
+
+You can import `.json` files and have them expanded to a JavaScript object:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/my-component-story-import-json.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+If you want to know the exact details of the webpack config, the best way is to run either of the following:
+
+```shell
+
+## Development mode
+yarn start-storybook --debug-webpack
+
+## Production mode
+yarn build-storybook --debug-webpack
+```
+
+### Bundle splitting
+
+Starting with Storybook 6.4, [bundle splitting](https://v4.webpack.js.org/guides/code-splitting/) is supported through a configuration flag. Update your Storybook configuration and add the `storyStoreV7` flag:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-on-demand-story-loading.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+When you start your Storybook, you'll see an improvement in loading times. Read more about it in the [announcement post](https://storybook.js.org/blog/storybook-on-demand-architecture/) and the [configuration documentation](../configure/overview.md#configure-your-storybook-project).
+
+### Webpack 5
+
+Storybook builds your project with Webpack 4 by default. If your project uses Webpack 5, you can opt into the Webpack 5 builder by installing the required dependencies (i.e., `@storybook/builder-webpack5`, `@storybook/manager-webpack5`) and update your Storybook configuration as follows:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-main-webpack5.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+Once you are using Webpack 5, you can further opt into some features to optimize your build:
+
+#### Lazy Compilation
+
+Storybook supports Webpack's experimental [lazy compilation](https://webpack.js.org/configuration/experiments/#experimentslazycompilation) feature, via the `lazyCompilation` builder flag:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-main-webpack5-lazyCompilation.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+This feature applies in development mode, and will mean your Storybook will start up faster, at the cost of slightly slower browsing time when you change stories.
+
+#### Filesystem Caching
+
+Storybook supports Webpack's [filesystem caching](https://webpack.js.org/configuration/cache/#cachetype) feature, via the `fsCache` builder flag:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-main-webpack5-fsCache.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+This feature will mean build output is cached between runs of Storybook, speeding up subsequent startup times.
+
+### Extending Storybook’s webpack config
```

Comment:
Please make sure all Webpack's are capitalized when used in a sentence or phrase